synapse/axon.py

```diff
@@ -242,13 +241,39 @@ async def __anit__(self, dirn, conf=None):  # type: ignore
         self.axonmetrics.setdefault('size:bytes', 0)
         self.axonmetrics.setdefault('file:count', 0)
 
+        self._initAxonLimits()
         self.addHealthFunc(self._axonHealth)
 
         # modularize blob storage
         await self._initBlobStor()
 
         self._initAxonHttpApi()
 
+    def _reqBelowLimit(self):
+
+        if (self.maxbytes is not None and
+            self.maxbytes <= self.axonmetrics.get('size:bytes')):
+            mesg = f'Axon is at size:bytes limit: {self.maxbytes}'
+            raise s_exc.HitLimit(mesg=mesg)
+
+        if (self.maxcount is not None and
+            self.maxcount <= self.axonmetrics.get('file:count')):
+            mesg = f'Axon is at file:count limit: {self.maxcount}'
+            raise s_exc.HitLimit(mesg=mesg)
+
+    def _initAxonLimits(self):
+
+        self.maxbytes = None
+        self.maxcount = None
+
+        maxcount = os.getenv('AXON_MAX_COUNT')
```

Comment:
Why not bake this into the cellconf ?

Reply:
Mostly because we dont want it to be very visible, but if you think it should be i guess we could.

Reply:
per discussionn, going to plumb this in as a hidden conf option.